src/uproot/_dask.py

```diff
@@ -1026,84 +1031,19 @@ def read_tree(
         assert tree.source  # we must be reading something here
         return out, tree.source.performance_counters
 
-    def mock(self) -> AwkArray:
-        awkward = uproot.extras.awkward()
-        return awkward.typetracer.typetracer_from_form(
-            self.expected_form,
-            highlevel=True,
-            behavior=self.form_mapping_info.behavior,
-        )
-
-    def mock_empty(self, backend) -> AwkArray:
-        awkward = uproot.extras.awkward()
-        return awkward.to_backend(
-            self.expected_form.length_zero_array(highlevel=False),
-            backend=backend,
-            highlevel=True,
-            behavior=self.form_mapping_info.behavior,
-        )
-
-    def prepare_for_projection(self) -> tuple[AwkArray, TypeTracerReport, dict]:
-        awkward = uproot.extras.awkward()
-        dask_awkward = uproot.extras.dask_awkward()
-
-        # A form mapping will (may) remap the base form into a new form
-        # The remapped form can be queried for structural information
-
-        # Build typetracer and associated report object
-        meta, report = awkward.typetracer.typetracer_with_report(
-            self.expected_form,
-            highlevel=True,
-            behavior=self.form_mapping_info.behavior,
-            buffer_key=self.form_mapping_info.buffer_key,
-        )
+    @property
+    def form(self):
+        return self.expected_form
 
-        return (
-            meta,
-            report,
-            {
-                "trace": dask_awkward.lib.utils.trace_form_structure(
-                    self.expected_form,
-                    buffer_key=self.form_mapping_info.buffer_key,
-                ),
-                "form_info": self.form_mapping_info,
-            },
-        )
+    def project(self, columns) -> T:
+        from dask_awkward.lib.utils import _buf_to_col
 
-    def project(self: T, *, report: TypeTracerReport, state: dict) -> T:
-        keys = self.necessary_columns(report=report, state=state)
+        keys = [_buf_to_col(c).replace(".", "_") for c in columns]
```

Comment:
@lgray : this is the simplistic way to unmap the column names. This line for the simple case, the if-block for nanoAOD, adding in implicit columns like nJet. Then, this doesn't actually make use of the schema and keys_for_buffer_keys

Reply:
(Reminder to @lgray to take a look at this.)